src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp

```diff
@@ -248,5 +248,226 @@ QVariantMap QgsLayoutAtlasToPdfAlgorithm::processAlgorithm( const QVariantMap &p
   return outputs;
 }
 
+// QgsLayoutAtlasToMultiplePdfAlgorithm
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::name() const
+{
+  return QStringLiteral( "atlaslayouttomultiplepdf" );
+}
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::displayName() const
+{
+  return QObject::tr( "Export atlas layout as PDF (multiple files)" );
+}
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Exports an atlas layout to multiple PDF files." );
+}
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm outputs an atlas layout to multiple PDF files.\n\n"
+                      "If a coverage layer is set, the selected layout's atlas settings exposed in this algorithm "
+                      "will be overwritten. In this case, an empty filter or sort by expression will turn those "
+                      "settings off.\n"
+                    );
+}
+
+void QgsLayoutAtlasToMultiplePdfAlgorithm::initAlgorithm( const QVariantMap & )
```

Comment:
there's a lot of near-identical code shared between these two algorithms. Can you refactor this out so that the common code is in a shared base class?